syntax = "proto3";

message Payload {
  enum Action {
    CREATE_PROJECT = 0;
    CREATE_ITEM = 1;
    EDIT_ITEM = 2;
    INCREMENT_SPRINT = 3;
    ADD_USER = 4;
    REMOVE_USER = 5;
  }

  // An action tells the handler which function
  // the payload requested to use.
  Action action = 1;

  // Time the transaction was sent.
  uint64 timestamp = 2;
  
  // All the data neccessitated by the actions
  // defined in the enum.
  CreateProjectAction create_project = 3;
  CreateItemAction create_item = 4;
  EditItemAction edit_item = 5;
  IncrementSprintAction increment_sprint = 6;
  AddUserAction add_user = 7;
  RemoveUserAction remove_user = 8;
}

message CreateProjectAction {
  //project name
  string project_name = 1;


}

message CreateItemAction {

}

message EditItemAction {

}

message IncrementSprintAction {

}

message AddUserAction {

}

message RemoveUserAction {

}